sql/table.go

```diff
@@ -199,23 +203,53 @@ func makeColumnDefDescs(d *parser.ColumnTableDef) (*ColumnDescriptor, *IndexDesc
 	return col, idx, nil
 }
 
-func (p *planner) getTableDesc(qname *parser.QualifiedName) (TableDescriptor, *roachpb.Error) {
+func getKeysForTableDescriptor(tableDesc *TableDescriptor) (
+	zoneKey roachpb.Key, nameKey roachpb.Key, descKey roachpb.Key) {
+
+	zoneKey = MakeZoneKey(tableDesc.ID)
+	nameKey = MakeNameMetadataKey(tableDesc.ParentID, tableDesc.GetName())
+	descKey = MakeDescMetadataKey(tableDesc.ID)
+	return
+}
+
+// getTableDescEx returns a table descriptor, or nil if the descriptor is not
+// found.
+func (p *planner) getTableDescEx(qname *parser.QualifiedName) (
```

Comment:
Why not mimick the interface of `p.getDescriptor`? Seems the right thing for the job.